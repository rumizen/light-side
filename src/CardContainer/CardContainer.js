import React from "react";
import "./CardContainer.scss";
import Card from '../Card/Card.js';

const CardContainer = props => {
  const allCards = props.category.map(thing => {
    return (
      <Card {...thing} updateFavorites={props.updateFavorites} />
    );
  });

  return (
    <main className="card-container">
      {allCards}
    </main>
  );
};

export default CardContainer;
