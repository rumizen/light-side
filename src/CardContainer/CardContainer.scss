.card-container {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  margin-top: 24px;
  animation: fader 1.5s;
}

@keyframes fader {
  from {opacity: 0;}
  to {opacity: 1;}
}