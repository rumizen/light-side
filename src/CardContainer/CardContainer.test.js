import React from "react";
import { shallow } from "enzyme";
import CardContainer from "./CardContainer";
import Data from "../mock-people";

describe("CardContainer", () => {
  it("should match the CardContainer snapshot", () => {
    const wrapper = shallow(<CardContainer category={Data.results} />);

    expect(wrapper).toMatchSnapshot();
  });

  it("should match the CardContainer snapshot with no favorites", () => {
    const mockData = [];
    const wrapper = shallow(<CardContainer category={mockData} />);

    expect(wrapper).toMatchSnapshot();
  });
});
