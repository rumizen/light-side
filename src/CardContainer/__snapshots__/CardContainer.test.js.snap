// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer should match the CardContainer snapshot 1`] = `ShallowWrapper {}`;

exports[`CardContainer should match the CardContainer snapshot with no favorites 1`] = `ShallowWrapper {}`;
